sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasQueryParameters.cs

```diff
@@ -0,0 +1,129 @@
+﻿//// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+
+namespace Azure.Storage.Sas
+{
+    /// <summary>
+    /// A <see cref="BlobSasQueryParameters"/> object represents the components
+    /// that make up an Azure Storage Shared Access Signature's query
+    /// parameters.  You can construct a new instance using
+    /// <see cref="BlobSasBuilder"/>.
+    /// 
+    /// For more information, <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas"/>.
+    /// </summary>
+    public sealed class BlobSasQueryParameters : SasQueryParameters
+    {
+        /// <summary>
+        /// Gets the Azure Active Directory object ID in GUID format.
+        /// </summary>
+        public new string KeyOid => base.KeyOid;
```

Comment:
I am confused with `Key` as a prefix for all these properties. The type is `BlobSasQueryParameters` but there isn't something pointing to Key... could u expand more on why we have it?

Reply:
Sure, it's how you make shared access signatures work with token credentials.  Generally you take all this SAS stuff, bundle it into a carefully formatted string, and then sign it with your shared key so the Storage service can authorize your request.  But if you're using token credentials, you don't have a shared key to sign with so Storage added a concept of creating delegation keys that are like temporary shared keys which forward on to your AD credentials.

Reply:
Thanks!